fix for flakey analyze test

eliasyishak opened the pull request:
This PR fixes #111272 by improving how we parse the stdout from `flutter analyze` to find only the relevant lines

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

eliasyishak committed:
regex pattern approach for flakey analyze test

eliasyishak committed:
adding logging for errors

eliasyishak committed:
regex pattern handles windows output

eliasyishak commented on packages/flutter_tools/test/commands.shard/permeable/create_test.dart:
```diff
@@ -40,6 +40,7 @@ const String _kNoPlatformsMessage = "You've created a plugin project that doesn'
 const String frameworkRevision = '12345678';
 const String frameworkChannel = 'omega';
 const String _kDisabledPlatformRequestedMessage = 'currently not supported on your local environment.';
+const String _kLintingErrorPattern = r"^\s*(?<type>error|info)(\s*[•|-]\s*)(?<desc>[\(\)`@\w\s'-]+)(\s*[•|-]\s*)(?<path>[\\\w'-\/:]+)(\s*[•|-]\s*)(?<lint>[\w'-]+)$";
```
Agreed, the pattern is hard to decipher. But I went with this approach because I wasn't sure if the `analyze` tool could potentially output more lines after the `Analyzing...` line.

I can make the necessary updates to only check lines after that particular line if you think that is better?

eliasyishak replied:
To help with deciphering it as well, I added named groups within the regex matches so it could potentially make it easier to debug... but adding the named groups also makes the pattern string pretty long

christopherfujino replied:
> Agreed, the pattern is hard to decipher. But I went with this approach because I wasn't sure if the analyze tool could potentially output more lines after the Analyzing... line.

"Analyzing flutter_project..." occurs after we're done with the pub get flow: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/analyze_once.dart#L115. It's not perfect, but at least if we regress this I'm pretty sure it will be in our own tool code.

eliasyishak committed:
removing difficult regex pattern

instead looking for the first instance of `Analyzing flutter_project...` to determine when to start capturing error outputs from `flutter analyze`

christopherfujino commented on packages/flutter_tools/test/commands.shard/permeable/create_test.dart:
```diff
@@ -3132,14 +3131,20 @@ Future<void> _analyzeProject(String workingDir, { List<String> expectedFailures
-        .toList();
+    bool analyzeLineFound = false;
+    const LineSplitter().convert(stdout).forEach((String line) {
+      if (!analyzeLineFound && line.startsWith('Analyzing flutter_project')) {
```
nit: can you add a comment explaining we want to ignore all output from pub get?

christopherfujino reviewed (approved):
LGTM

eliasyishak committed:
trimming whitespace + including comment for conditional

eliasyishak committed:
removing hard coded flutter_project name

eliasyishak committed:
clean up trailing whitespace
